volttron/platform/certs.py

```diff
@@ -1022,3 +1022,182 @@ def create_root_ca(self, overwrite=True, valid_days=DEFAULT_DAYS, **kwargs):
 
         self._save_cert(self.root_ca_name, cert, pk)
         self.rebuild_requests_ca_bundle()
+
+
+def _create_private_key():
+    return rsa.generate_private_key(
+        public_exponent=65537,
+        key_size=2048,
+        backend=default_backend()
+    )
+
+
+def _create_signed_certificate(ca_cert, ca_key, name,
```

Comment:
Why does repeat all the logic in create_ca_signed_certificate. Can we update parameters in create_ca_signed_certificated instead?

Reply:
I think that I was headed to moving the create signed certificate out of the class into an _function....then the certs class would be responsible for storing it in a file.

Reply:
I was actually in favor of making create_ca_signed_certificate use this rather than having it in the Certs object.  Since this will return the cert and key then we just save it in the other function.